test/unit/org/apache/cassandra/service/StartupChecksTest.java

```diff
@@ -66,6 +76,11 @@ public void setup() throws IOException
         Files.createDirectories(sstableDir);
 
         startupChecks = new StartupChecks();
+
+        // Create and attach an in-memory ListAppender to capture warnings logged
+        testAppender = new ListAppender<>();
+        testAppender.start();
+        ((Logger) LoggerFactory.getLogger(StartupChecks.class)).addAppender(testAppender);
     }
 
     @After
```

Comment:
I forgot to mention we should clear the logs from the appender after each test, otherwise we carry them over to the next test case. Sthg that can be done in some other ticket.